### PR TITLE
KAFKA-15623: Remove junit 4 from  stream module 

### PR DESCRIPTION
**Pull request by m1a2st**
remove all junit 4 dependency from stream module

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -509,18 +509,9 @@ subprojects {
...
-        // junit-vintage (JUnit 4) can be removed once the JUnit 4 migration is complete.
-        includeEngines "junit-vintage", "junit-jupiter"
-      }
-    } else {
-      useJUnitPlatform {
-        includeTags "integration"
-      }
+    useJUnitPlatform {
+      includeTags "integration"
+      includeEngines "junit-jupiter"
```
Comment (chia7712) [inline build.gradle]: Could you add it to `test` task for consistency?

File: build.gradle [hunk 2]
```diff
@@ -2305,7 +2287,6 @@ project(':streams') {
     testImplementation project(':server')
     testImplementation libs.reload4j
     testImplementation libs.junitJupiter
-    testImplementation libs.junitVintageEngine
```
Comment (chia7712) [inline build.gradle:2308]: please remove it from `dependencies.gradle` too

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for this patch

**Comment by m1a2st** [comment]
@chia7712, Thanks for your conversations, PTAL

File: build.gradle [hunk 3]
```diff
@@ -482,7 +482,10 @@ subprojects {
 
     exclude testsToExclude
 
-    useJUnitPlatform()
+    useJUnitPlatform {
+      includeTags "integration"
```
Comment (chia7712) [inline build.gradle]: this is incorrect. we have three test-related tasks: `test`, `unitTest` and `integrationTest`. `test` does not need to include/exclude tags. `unitTest` needs `excludeTags "integration"` and `integrationTest` needs `includeTags "integration"`

File: build.gradle [hunk 4]
```diff
@@ -482,7 +482,9 @@ subprojects {
 
     exclude testsToExclude
 
-    useJUnitPlatform()
+    useJUnitPlatform {
+      includeEngines "junit-jupiter"
```
Comment (chia7712) [inline build.gradle]: `includeEngines 'junit-jupiter'`

File: build.gradle [hunk 5]
```diff
@@ -509,18 +511,8 @@ subprojects {
...
-        // Both engines are needed to run JUnit 4 tests alongside JUnit 5 tests.
-        // junit-vintage (JUnit 4) can be removed once the JUnit 4 migration is complete.
-        includeEngines "junit-vintage", "junit-jupiter"
-      }
-    } else {
-      useJUnitPlatform {
-        includeTags "integration"
-      }
+    useJUnitPlatform {
+      includeTags "integration"
```
Comment (chia7712) [inline build.gradle:515]: Could you please add specific test engine explicitly?

File: build.gradle [hunk 6]
```diff
@@ -546,18 +538,8 @@ subprojects {
...
-        // Both engines are needed to run JUnit 4 tests alongside JUnit 5 tests.
-        // junit-vintage (JUnit 4) can be removed once the JUnit 4 migration is complete.
-        includeEngines "junit-vintage", "junit-jupiter"
-      }
-    } else {
-      useJUnitPlatform {
-        excludeTags "integration"
-      }
+    useJUnitPlatform {
+      excludeTags "integration"
```
Comment (chia7712) [inline build.gradle:543]: Could you please add specific test engine explicitly?

**Review by chia7712 — COMMENTED** [review]
@m1a2st thanks for updated PR.

BTW, could you please remove the duplicate declaration from `storage` module?

https://github.com/apache/kafka/blob/trunk/build.gradle#L2031

**Pull request merged**